content/advent-2017/using-go-templates.md

```diff
@@ -0,0 +1,315 @@
+
+Using the `range` action you can loop through a slice. A range actions is defined using the `{{range .Member}} ... {{end}}` template.
+
+If your slice is non-struct type, you can refer to the value using the `{{ . }}` action. In case of structs, you can refer to the value using the `{{ .Member }}` action, as already explained.
+
+#### Functions, Pipelines and Variables
+
+Actions have several built-in functions that're used along with pipelines to additionally parse output. Pipelines are annotated with `|` and default behavior is sending data from left side to the function on right side.
+
+Functions are used to escape the action's result. There're several functions available by default such as, `html` which returns HTML escaped output, safe against code injection or `js` which returns JavaScript escaped output.
+
+Using the `with` action, you can define variables that're available in that `with` block: `{{ with $x := <^>result-of-some-action<^> }} {{ $x }} {{ end }}`.
+
+Throughput the article, we're going to cover more complex actions, such as reading from array instead of struct.
+
+## Parsing Templates
+
+The three most important and most frequently used functions are:
+
+* `New` — allocates new, undefinied template,
+* `Parse` — parses given template string and return parsed template,
+* `Execute` — applies parsed template to the data structure and write result to the given writer.
+
+The following code shows above-mentioned functions in the action:
+
+```go
+package main
+
+import (
+	"os"
+	"text/template"
+)
+
+type Todo struct {
+	Name        string
+	Description string
+}
+
+func main() {
+	td := Todo{"Test templates", "Let's test an template to see the magic."}
+
+  t, err := template.New("todos").Parse("You have task named \"{{ .Name}}\" with description: \"{{ .Description}}\"")
+	if err != nil {
+		panic(err)
+	}
+	err = t.Execute(os.Stdout, td)
+	if err != nil {
+		panic(err)
+	}
+}
+```
+
+The result is the following message printed in your terminal:
+
+```
+You have task named "Test templates" with description: "Let's test an template to see the magic."
+```
+
+You can reuse the same template, without needing to create or parse it again by providing the struct you want to use to the `Execute` function again:
+
+```go
+// code omitted beacuse of brevity
+...
+
+tdNew := Todo{"Go", "Contribute to any Go project"}
+err = t.Execute(os.Stdout, tdNew)
+}
+```
+
+The result is like the previous one, just with new data:
+
+```
+You have task named "Go" with description: "Contribute to any Go project"
+```
+
+As you can see, templates provide powerful way to customize textual output. Beside manipulating textual output, you can also manipulate HTML output using the `html/template` package.
+
+### Verifying Templates
+
+`template` packages provide the `Must` functions, used to verify is template correct on parsing time. `Must` function provides the same result as if we manually checked for the error, like in the previous example. 
```

Comment:
s/used to verify is template correct on parsing time/used to verify that a template is correct during parsing (more than one option work here)